api/event.js

```diff
@@ -0,0 +1,88 @@
+const fetch = require('node-fetch')
+const gqlr = require('graphql-request')
+const { GraphQLClient } = gqlr
+const { API_URL } = require('../src/config')
+
+const EndPoint = `${API_URL}/graphql`
+const GetParty = `
+  query getParty($address: String!) {
+    party(address: $address) {
+      id
+      name
+      description
+      headerImg
+      address
+    }
+  }
+`
+
+function htmlEncode(str) {
+  return str
+    ? str
+        .replace(/\&/g, '&amp;')
+        .replace(/\>/g, '&gt;')
+        .replace(/\</g, '&lt;')
+        .replace(/\"/g, '&quot;')
+    : ''
+}
+
+module.exports = async (req, res) => {
+  const paths = req.url.split('/')
+  const address = paths[paths.length - 1]
+
+  // fetch event info via graphql
+  const client = new GraphQLClient(EndPoint, {
+    headers: {
+      Authorization: ``
+    }
+  })
+  const { party } = await client.request(GetParty, { address })
+
+  // fetch index.html and replace placeholders
+  const baseUrl = `https://${req.headers.host}`
+  const r = await fetch(`${baseUrl}/build/index.html`)
+  if (r.ok) {
```

Comment:
We should probably handle a default/error response if the response is not `.ok`

Reply:
yeah. I thought about that but didn't get time to look at the current practice in the app. 

Could @makoto or @jefflau recommend the preferred error message to be displayed here if no `index.html` is fetched, otherwise a 502 error (actually timeout after 10s in the serverless function) will be thrown.